### ISSUE TITLE
[branched][rawhide][s390x] selinux denials cause `systemd-ssh-generator` to fail

### ISSUE THREAD

**Issue by jbtrystram** [coreos/fedora-coreos-tracker#1786]
i am still seeing this errors on the s390x build of branched : https://jenkins-fedora-coreos-pipeline.apps.ocp.fedoraproject.org/blue/organizations/jenkins/build-arch/detail/build-arch/106/pipeline/

relevant packages : 
selinux-policy 40.13-1.fc40 -> 41.14-1.fc41
selinux-policy-targeted 40.13-1.fc40 -> 41.14-1.fc41
systemd 255.3-1.fc40 -> 256.5-1.fc41

Most tests fails with
```
[    8.131680] (sd-exec-[1340]: /usr/lib/systemd/system-generators/systemd-ssh-generator failed with exit status 1.
```

Relevant selinux denials  : 
```
[    8.111974] audit: type=1400 audit(1724677759.011:4): avc:  denied  { read } for  pid=1358 comm="systemd-ssh-gen" name="sysinfo" dev="proc" ino=4026531945 scontext=system_u:system_r:systemd_ssh_generator_t:s0 tcontext=system_u:object_r:sysctl_t:s0 tclass=file permissive=0
```

**Comment by dustymabe** [comment]
I assume since the `permissive=0` only on one of them the culprit here is: 

```
[    8.111974] audit: type=1400 audit(1724677759.011:4): avc:  denied  { read } for  pid=1358 comm="systemd-ssh-gen" name="sysinfo" dev="proc" ino=4026531945 scontext=system_u:system_r:systemd_ssh_generator_t:s0 tcontext=system_u:object_r:sysctl_t:s0 tclass=file permissive=0
```

**Comment by jbtrystram** [comment]
I failed to find the relevant issue earlier but this looks like the same as https://github.com/coreos/fedora-coreos-tracker/issues/1735

**Comment by dustymabe** [comment]
ok I think it's now a bit more clear about why we are only seeing this on s390x. The `/proc/sysinfo` file only seems to exist on s390x.

```
Fedora CoreOS 41.20240827.92.0
Tracker: https://github.com/coreos/fedora-coreos-tracker
Discuss: https://discussion.fedoraproject.org/tag/coreos

Last login: Tue Aug 27 14:35:25 2024
[core@cosa-devsh ~]$ 
[core@cosa-devsh ~]$ arch
x86_64
[core@cosa-devsh ~]$ stat /proc/sysinfo
stat: cannot statx '/proc/sysinfo': No such file or directory
```

```
Fedora CoreOS 41.20240827.92.0
[core@cosa-devsh ~]$ 
[core@cosa-devsh ~]$ arch
s390x
[core@cosa-devsh ~]$ stat /proc/sysinfo
  File: /proc/sysinfo
  Size: 0               Blocks: 0          IO Block: 1024   regular empty file
Device: 0,21    Inode: 4026531945  Links: 1
Access: (0444/-r--r--r--)  Uid: (    0/    root)   Gid: (    0/    root)
Context: system_u:object_r:sysctl_t:s0
Access: 2024-08-27 14:38:06.987579917 +0000
Modify: 2024-08-27 14:38:06.987579917 +0000
Change: 2024-08-27 14:38:06.987579917 +0000
 Birth: -
```

**Comment by dustymabe** [comment]
> I failed to find the relevant issue earlier but this looks like the same as #1735

That one is `comm="systemd-ssh-gen" name="vsock" dev="devtmpfs"` - so not the same failure as this one as it is trying to access a file named `vsock` and not `sysinfo`.

**Comment by jbtrystram** [comment]
I am filing a RHBZ with details

**Comment by jlebon** [comment]
Likely https://github.com/systemd/systemd/blob/9c0aee7cbbc82aa030ba9c78af1e7fd0ff520d27/src/ssh-generator/ssh-generator.c#L197 --> ... --> https://github.com/systemd/systemd/blob/9c0aee7cbbc82aa030ba9c78af1e7fd0ff520d27/src/basic/virt.c#L424-L445 where we clearly see the s390x-specific bits.

**Comment by dustymabe** [comment]
upstream fix merged: https://github.com/fedora-selinux/selinux-policy/pull/2326

**Comment by travier** [comment]
F41 freeze excception: https://bugzilla.redhat.com/show_bug.cgi?id=2308177

**Comment by dustymabe** [comment]
This was fixed in `selinux-policy-41.15-1` which is already in rawhide and fast-tracked in `branched`.

These two PRs close this issue out:

- https://github.com/coreos/fedora-coreos-config/pull/3133
- https://github.com/coreos/fedora-coreos-config/pull/3131
